share/availability/light/availability_test.go

```diff
@@ -399,38 +384,111 @@ func TestPrunePartialFailed(t *testing.T) {
 	require.False(t, exist)
 }
 
-var _ exchange.SessionExchange = (*fakeSessionExchange)(nil)
+func TestPruneWithCancelledContext(t *testing.T) {
```

Comment:
Addition of this test and related refactoring looks unrelated to PR, but ok

Reply:
It's related cause we want be sure that cancelled context works too. (This PR isn't about refactoring from the start LOL).